lib/js/src/manager/screen/_SubscribeButtonManagerBase.js

```diff
@@ -0,0 +1,189 @@
+            await this._subscribeButtonRequest(buttonName, listener);
+            return;
+        }
+
+        if (this._onButtonListeners.get(buttonName).includes(listener)) {
+            console.info(`Already subscribed to the button named ${buttonName}`);
+            return;
+        }
+
+        this._onButtonListeners.get(buttonName).push(listener);
+    }
+
+    /**
+     * Removes a listener from the list of listeners and unsubscribes to the button if it was the last listener.
+     * @param {ButtonName} buttonName - Name of the button
+     * @param {Function} listener - The listener to be removed
+     */
+    async _removeButtonListener (buttonName, listener) {
+        if (listener === null || listener === undefined || buttonName === null || buttonName === undefined) {
+            console.error('ButtonName and OnButtonListener cannot be null or undefined');
+            return;
+        }
+
+        const listenerArray = this._onButtonListeners.get(buttonName);
+        if (listenerArray === null || listenerArray === undefined || !listenerArray.includes(listener)) {
+            console.error(`Attempting to unsubscribe to the ${buttonName} button failed because it is not currently subscribed`);
+            return;
+        }
+
+        if (this._onButtonListeners.get(buttonName).size() > 1) {
+            this._onButtonListeners.set(buttonName, _ArrayTools.arrayRemove(listenerArray, listener));
+            return;
+        }
+        await this._unsubscribeButtonRequest(buttonName, listener);
+    }
+
+    /**
+     * Sends a SubscribeButton RPC and adds a listener to the list of listeners
+     * @param {ButtonName} buttonName - Name of the button
+     * @param {Function} listener - A function to invoke when a button notification occurs
+     */
+    async _subscribeButtonRequest (buttonName, listener) {
+        const subscribeButton = new SubscribeButton().setButtonName(buttonName);
+        const response = await this._lifecycleManager.sendRpcResolve(subscribeButton);
+        if (response.getSuccess()) {
+            let listenerArray = this._onButtonListeners.get(buttonName);
+            // If no array exists yet for this function id, create one
+            if (!Array.isArray(listenerArray)) {
+                listenerArray = [];
+            }
+            listenerArray.push(listener);
+            this._onButtonListeners.set(buttonName, listenerArray);
+        }
+    }
+
+    /**
+     * Sends an UnsubscribeButton RPC and removes a listener from the list of listeners
+     * @param {*} buttonName - Name of the button
+     * @param {*} listener - The listener to be removed
+     */
+    async _unsubscribeButtonRequest (buttonName, listener) {
+        const unsubscribeButton = new UnsubscribeButton().setButtonName(buttonName);
+        const response = await this._lifecycleManager.sendRpcResolve(unsubscribeButton);
+        if (response.getSuccess()) {
```

Comment:
Missing the `onError` case where the response isn't a success. Since this is an async function and we don't have a listener class to handle the error case, my recommendation is to throw a new Error object on failure with the same message the java version sends, and then the calling class can hook up a `.catch()` handler to receive those errors. Same issue for the `_subscribeButtonRequest` method